src/picongpu/include/particles/manipulators/RandomPositionImpl.hpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2015 Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#include "simulation_defines.hpp"
+#include "nvidia/rng/RNG.hpp"
+#include "nvidia/rng/methods/Xor.hpp"
+#include "nvidia/rng/distributions/Uniform_float.hpp"
+#include "mpi/SeedPerRank.hpp"
+
+namespace picongpu
+{
+namespace particles
+{
+namespace manipulators
+{
+
+namespace nvrng = nvidia::rng;
+namespace rngMethods = nvidia::rng::methods;
+namespace rngDistributions = nvidia::rng::distributions;
+
+template< typename T_SpeciesType>
+struct RandomPositionImpl
+{
+    typedef T_SpeciesType SpeciesType;
+    typedef typename MakeIdentifier<SpeciesType>::type SpeciesName;
+
+    HINLINE RandomPositionImpl(uint32_t currentStep) : isInitialized(false)
+    {
+        typedef typename SpeciesType::FrameType FrameType;
+
+        GlobalSeed globalSeed;
+        mpi::SeedPerRank<simDim> seedPerRank;
+        seed = seedPerRank(globalSeed(), FrameType::CommunicationTag);
+        seed ^= POSITION_SEED ^ currentStep;
```

Comment:
uh, that is the point where we need to introduce a per-species seed, else it will stay the same position ;)